src/connector/src/sink/file_sink/batching_log_sink.rs

```diff
@@ -0,0 +1,147 @@
+                        epoch,
+                        curr_epoch
+                    );
+                    LogConsumerState::EpochBegun { curr_epoch: epoch }
+                }
+                LogConsumerState::BarrierReceived { prev_epoch } => {
+                    assert!(
+                        epoch > prev_epoch,
+                        "new epoch {} should be greater than prev epoch {}",
+                        epoch,
+                        prev_epoch
+                    );
+                    LogConsumerState::EpochBegun { curr_epoch: epoch }
+                }
+            };
+            match item {
+                LogStoreReadItem::StreamChunk { chunk, chunk_id } => {
+                    sink_writer.write_batch(chunk, chunk_id).await?;
+                    match sink_writer.try_commit().await {
+                        Err(e) => {
+                            return Err(e);
+                        }
+                        // The file has been successfully written and is now visible to downstream consumers.
+                        // Truncate the file to remove the specified `chunk_id` and any preceding content.
+                        Ok(Some(chunk_id)) => {
+                            log_reader.truncate(TruncateOffset::Chunk {
+                                epoch: (epoch),
+                                chunk_id: (chunk_id),
+                            })?;
+                        }
+                        // The file has not been written into downstream file system.
+                        Ok(None) => {}
+                    }
+                }
+                LogStoreReadItem::Barrier { is_checkpoint: _ } => {
+                    let prev_epoch = match state {
+                        LogConsumerState::EpochBegun { curr_epoch } => curr_epoch,
+                        _ => unreachable!("epoch must have begun before handling barrier"),
+                    };
+
+                    // The current sink specifies a batching strategy, which means that sink decoupling is enabled; therefore, there is no need to forcibly write to the file when the checkpoint barrier arrives.
+                    // When the barrier arrives, call the writer's try_finish interface to check if the file write can be completed.
+                    // If it is completed, which means the file is visible in the downstream file system, thentruncate the file in the log store; otherwise, do nothing.
+                    if let Some(committed_chunk_id) = sink_writer.try_commit().await? {
+                        // If epoch increased, we first need to truncate the previous epoch.
+                        if epoch > prev_epoch {
+                            log_reader.truncate(TruncateOffset::Barrier { epoch: prev_epoch })?;
+                        }
+                        log_reader.truncate(TruncateOffset::Chunk {
+                            epoch: (epoch),
+                            chunk_id: (committed_chunk_id),
```

Comment:
Is the commited chunk id always the last seen chunk id in batch log sinker? I wonder whether it is clearer to store the chunk id as part of the LogConsumeState instead of in the writer. The current implementation is correct but can be prone to bugs in the future because chunk id is only unique within a epoch but the writer currently only store the chunk id without epoch.

Reply:
Discussed offline, we don't need to store `chunk_id` in both `LogConsumeState` and `SinkWriter`:
- we only need to truncate `TruncateOffset::Chunk ` when LogStoreReadItem is StreamChunk, here we can get the chunk_id
- When barrier arrives, we only need to truncate `TruncateOffset::Barrier(prev_epoch)`, which is equal to truncate all the data before new barrier.